libslide/src/bignum/compare.rs

```diff
@@ -0,0 +1,71 @@
+use crate::bignum::Bignum;
+use std::cmp::Ordering;
+
+impl Ord for Bignum {
+    fn cmp(&self, other: &Bignum) -> Ordering {
+        match self.int.len().cmp(&other.int.len()) {
+            Ordering::Greater => return Ordering::Greater,
+            Ordering::Less => return Ordering::Less,
+            _ => (),
+        }
+
+        for i in (0..other.int.len()).rev() {
+            match self.int[i].cmp(&other.int[i]) {
+                Ordering::Greater => return Ordering::Greater,
+                Ordering::Less => return Ordering::Less,
+                _ => (),
+            }
+        }
+
+        let smaller_dec_size: usize = if self.dec.len() > other.dec.len() {
+            other.dec.len()
+        } else {
+            self.dec.len()
+        };
+
+        for i in 0..smaller_dec_size {
+            match self.dec[i].cmp(&other.dec[i]) {
+                Ordering::Greater => return Ordering::Greater,
+                Ordering::Less => return Ordering::Less,
+                _ => (),
+            }
+        }
+
+        match other.dec.len().cmp(&self.dec.len()) {
+            Ordering::Equal => Ordering::Equal,
+            Ordering::Less => Ordering::Less,
+            Ordering::Greater => Ordering::Greater,
+        }
+    }
+}
+
+impl Eq for Bignum {}
+
+impl PartialEq for Bignum {
+    fn eq(&self, other: &Bignum) -> bool {
+        if self.int.len() != other.int.len() {
+            return false;
+        }
+        for i in 0..other.int.len() {
+            if other.int[i] != self.int[i] {
+                return false;
+            }
+        }
+        if self.dec.len() != other.dec.len() {
+            return false;
+        }
+        for i in 0..other.dec.len() {
+            if other.dec[i] != self.dec[i] {
+                return false;
+            }
+        }
+        true
```

Comment:
Why not just

```rust
self.cmp(other) == Ordering::Equal
```

Reply:
What do you mean

Reply:
Replace this whole method with `self.cmp(other) == Ordering::Equal`